### DISCUSSION TITLE
Alternate input methods

### DISCUSSION THREAD

**Discussion by ChefDaddyJay** [jdwilkin4/contrast-ratio-repo#152] [Ideas]
As of writing, the application supports text input representing RGB, HSL, HEX and named colors. Some other methods might include:

- A color wheel
- A color picker
- A suggestions pallet (either a static display or perhaps something that responds to the user's input to either field)
- Sliders for RGB and HSL values

I'm sure there are more. Some of these may be more complicated than others.

**Comment by ChefDaddyJay** [comment]
Continuing this discussion, specifically on the idea of the color picker, I see two directions we could take it: 

1. Inspired by [Google's color picker](https://www.google.com/search?q=color+picker&rlz=1CAEVJI_enUS1041&oq=color+picker&aqs=chrome..69i57j0i512l2j0i433i512l4j5.1953j0j4&sourceid=chrome&ie=UTF-8) (pictured below), we could use sliders for each input, allowing the user to choose two points on the color spectrum to compare. This is the simpler of the two options and could allow us to integrate some of the other ideas mentioned above (pallet suggestions, brightness slider, etc.) pretty easily in the future. 

[![Google color picker utility](https://user-images.githubusercontent.com/97722492/226148962-0ccb4dff-63e1-4daf-a395-a867ce48f058.png)](https://www.google.com/search?q=color+picker&rlz=1CAEVJI_enUS1041&oq=color+picker&aqs=chrome..69i57j0i512l2j0i433i512l4j5.1953j0j4&sourceid=chrome&ie=UTF-8)

2. A somewhat more complicated option would be to allow the user to select a color from an image. Below is a screenshot of [imagecolorpicker.com](https://imagecolorpicker.com/en). Their application goes a bit outside the scope of this project by providing a full color pallet based on the given image, etc. Ours might instead identify the lowest and highest contrast points of a given image. Still, working with these images will add a new layer of complexity and might be better left for a later version.

[![imagecolorpicker.com color picker utility](https://user-images.githubusercontent.com/97722492/226153697-01ddefb8-b163-4800-822e-4b9a63dd0ae3.png)](https://imagecolorpicker.com/en)

For now, I'll be discussing the first option. More discussion on the second is, of course, welcome. Either way, the goal is to allow the user to select and compare colors from a given source, as opposed to text input. This is going to require a redesign of the current interface, which only accepts text. The new design should include the following:

- Text input to maintain current functionality - the app should continue to accept any input that it does currently (HEX, HSL, RGB)
- A drop-down for the named colors - this will help avoid things like misspellings, typos, etc. and give the user a list of colors to choose from, rather than leaving them to guess.
- A color selection pallet - this could take the form of a single slider displaying the full spectrum of colors combined with a pair of radio buttons to indicate whether the input is for the foreground or background color. A smaller slider could be used for each color independently, but this may take up more space and be less visually attractive. 
- A swatch to indicate the colors that have been selected - this can be separated from the input fields on the page so that it displays the user's selection regardless of input method and should include both large and small text to act as samples for the user.
- The calculated contrast ratio of the user's selection, along with the pass/fail rating for each standard - this information could be displayed next to the color swatches to link them visually with their respective samples.
- Finally, all other information currently on the page ("How does it work?", etc.) should be maintained and integrated into the new design.

While this sounds like a major change, the scope of the project remains the same: take two color inputs, calculate the contrast ratio, and display it to the user. This functionality is already in place. All this update should do is offer the user a more intuitive way to input those colors. 

I will try to put together a mock-up of this design, but if anyone else would like to take a stab at it, feel free. Also, any ideas, suggestions, or points I might have missed are welcome. Once we have a better picture of how this new interface will look, we can more easily break the task into individual issues to start working on.

One final note: we are adding a few items to the page, so we want to be careful not to overcrowd it. Readability is at the heart of this project, and we want to make sure that shows in the new design.

**Reply by Dorfieeee** [reply]
I'm for the option 1. We're building a color contrast ratio app. The color suggestion feature can be part of some further extension.
The Google color picker looks good. We can try to reverse engineer it. We need to think about two designs. One for a desktop app and one for a mobile.
This feature is a small project itself so it requires some considerations of how it is going to be implemented.
I think we could introduce some OOP and build it as a class.
